doc/source/_templates/defindex.html

```diff
@@ -0,0 +1,35 @@
+{#
+    basic/defindex.html
+    ~~~~~~~~~~~~~~~~~~~
+
+    Default template for the "index" page.
+
+    :copyright: Copyright 2007-2017 by the Sphinx team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+#}
+{%- extends "layout.html" %}
```

Comment:
Is this correct? First, there is the `-`, which I see nowhere in the jinja2 documentation, and second, I see that the other templates have a `!` preceding the path, which I assume means the base template is in another directory in the search path.

Reply:
So I'm thinking that perhaps this should be `{% extends "!layout.html" %}`, as our template library also includes `layout.html`.

Reply:
The `-` characters control deletion of the whitespace.

Reply:
The exlamation mark means to load the file from the template rather than from this template.
It's probably not needed here, since we are not including it from the same file.